### ISSUE TITLE
 Can you provide a script for inferring images

### ISSUE THREAD

**Issue by lfreee** [MaxZanella/CLIP-LoRA#4]
Can you provide code for predicting individual images？

**Comment by MaxZanella** [comment]
Hello,

You might find the CLIP basic example on their repository (https://github.com/openai/CLIP) helpful for your implementation:

import torch
import clip
from PIL import Image

device = "cuda" if torch.cuda.is_available() else "cpu"
model, preprocess = clip.load("ViT-B/32", device=device)

Insert your LoRA here with apply_lora.py and load_lora.py

image = preprocess(Image.open("CLIP.png")).unsqueeze(0).to(device)
text = clip.tokenize(["a diagram", "a dog", "a cat"]).to(device)

with torch.no_grad():
    image_features = model.encode_image(image)
    text_features = model.encode_text(text)
    logits_per_image, logits_per_text = model(image, text)
    probs = logits_per_image.softmax(dim=-1).cpu().numpy()

print("Label probs:", probs)  # prints: [[0.9927937  0.00421068 0.00299572]]

To integrate LoRA, you can utilize the apply_lora and load_lora functions from loralib/utils.py. These functions are also used in the run_lora function within the lora.py script if you want a more practical example. I hope this helps.

**Comment by lfreee** [comment]
I have already achieved it myself. Thanks.
